service/src/main/java/tds/assessment/services/impl/AssessmentServiceImpl.java

```diff
@@ -1,25 +1,78 @@
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import tds.assessment.Assessment;
+import tds.assessment.Form;
+import tds.assessment.Item;
+import tds.assessment.ItemConstraint;
+import tds.assessment.ItemProperty;
+import tds.assessment.Segment;
 import tds.assessment.repositories.AssessmentQueryRepository;
+import tds.assessment.repositories.FormQueryRepository;
+import tds.assessment.repositories.ItemQueryRepository;
 import tds.assessment.services.AssessmentService;
 
 @Service
 class AssessmentServiceImpl implements AssessmentService {
     private final AssessmentQueryRepository assessmentQueryRepository;
+    private final ItemQueryRepository itemQueryRepository;
+    private final FormQueryRepository formQueryRepository;
 
     @Autowired
-    public AssessmentServiceImpl(AssessmentQueryRepository assessmentQueryRepository) {
+    public AssessmentServiceImpl(AssessmentQueryRepository assessmentQueryRepository,
+                                 ItemQueryRepository itemQueryRepository,
+                                 FormQueryRepository formQueryRepository) {
+        this.formQueryRepository = formQueryRepository;
         this.assessmentQueryRepository = assessmentQueryRepository;
+        this.itemQueryRepository = itemQueryRepository;
     }
 
     @Override
     public Optional<Assessment> findAssessmentByKey(String assessmentKey) {
-        return assessmentQueryRepository.findAssessmentByKey(assessmentKey);
+        Optional<Assessment> maybeAssessment = assessmentQueryRepository.findAssessmentByKey(assessmentKey);
+
+        if (maybeAssessment.isPresent()) {
+            List<Form> forms = formQueryRepository.findFormsForAssessment(assessmentKey);
+            List<Item> items = itemQueryRepository.findItemsForAssessment(assessmentKey);
+            List<ItemProperty> itemProperties = itemQueryRepository.findActiveItemsProperties(assessmentKey);
+            List<ItemConstraint> itemConstraints = itemQueryRepository.findItemConstraintsForAssessment(maybeAssessment.get().getAssessmentId());
+            maybeAssessment.get().setItemConstraints(itemConstraints);
+
+            mapItems(items, itemProperties);
+            mapAssessment(maybeAssessment, forms, items);
+        }
+
+        return maybeAssessment;
+    }
+
+    private static void mapAssessment(Optional<Assessment> assessment, List<Form> forms, List<Item> items) {
+        for (Segment segment : assessment.get().getSegments()) {
+            segment.setForms(forms.stream()
+                .filter(form -> form.getSegmentKey().equals(segment.getKey())).collect(Collectors.toList()));
+            segment.setItems(items.stream()
+                .filter(item -> item.getSegmentKey().equals(segment.getKey())).collect(Collectors.toList()));
+        }
+    }
+
+    private static void mapItems(List<Item> items, List<ItemProperty> itemProperties) {
+        Multimap<String, ItemProperty> itemIdToPropertyMap = ArrayListMultimap.create();
```

Comment:
In hindsight, instead of using a Multimap to map the properties to each item, we could add an "itemId" property to the ItemProperty to link it back to its parent Item. Then we can use lambdas to map properties to the Items as we do with Forms and Items in lines 59-62 of this same file.